Scripts/Runtime/Entities/AbstractAnvilSystemBase.cs

```diff
@@ -49,5 +55,31 @@ protected new JobHandle Dependency
         /// Creates a new <see cref="AbstractAnvilSystemBase"/> instance.
         /// </summary>
         protected AbstractAnvilSystemBase() : base() { }
+
+        protected override void OnCreate()
+        {
+            base.OnCreate();
+            
+            //We could be created for a different world in which case we won't be in the group's update loop. 
+            //First we try to get the group attribute
+            Type type = GetType();
+            UpdateInGroupAttribute updateInGroupAttribute = type.GetCustomAttribute<UpdateInGroupAttribute>();
+            if (updateInGroupAttribute == null)
+            {
+                //If nothing, then we don't do anything, the developer is responsible
+                return;
+            }
+            
+            //We'll get or create the system specified in the attribute. It SHOULD be a Group, but might not be so...
+            ComponentSystemBase componentSystemBase = World.GetOrCreateSystem(updateInGroupAttribute.GroupType);
+            if (componentSystemBase is not ComponentSystemGroup componentSystemGroup)
+            {
+                //We'll rightly complain about it.
+                throw new InvalidOperationException($"{type.GetReadableName()} is trying to set its {nameof(UpdateInGroupAttribute)} to use {updateInGroupAttribute.GroupType} but that isn't a {typeof(ComponentSystemGroup).GetReadableName()}!");
+            }
+            //This function early returns if we're already part of the group, like in the case of automatic 
+            //default world init, so it's safe to just call to ensure we're part of the right update system.
+            componentSystemGroup.AddSystemToUpdateList(this);
```

Comment:
Looks good. Can we just put this in its own method?

Reply:
Yep, done!